src/renderer/renderElement.ts

```diff
@@ -7,154 +7,255 @@ import {
 } from "../element/bounds";
 import { RoughCanvas } from "roughjs/bin/canvas";
 import { Drawable } from "roughjs/bin/core";
+import { RoughSVG } from "roughjs/bin/svg";
+import { RoughGenerator } from "roughjs/bin/generator";
+import { SVG_NS } from "../utils";
 
-export function renderElement(
+function generateElement(
   element: ExcalidrawElement,
-  rc: RoughCanvas,
-  context: CanvasRenderingContext2D,
+  generator: RoughGenerator,
 ) {
-  const generator = rc.generator;
-  if (element.type === "selection") {
-    const fillStyle = context.fillStyle;
-    context.fillStyle = "rgba(0, 0, 255, 0.10)";
-    context.fillRect(0, 0, element.width, element.height);
-    context.fillStyle = fillStyle;
-  } else if (element.type === "rectangle") {
-    if (!element.shape) {
-      element.shape = generator.rectangle(0, 0, element.width, element.height, {
-        stroke: element.strokeColor,
-        fill:
-          element.backgroundColor === "transparent"
-            ? undefined
-            : element.backgroundColor,
-        fillStyle: element.fillStyle,
-        strokeWidth: element.strokeWidth,
-        roughness: element.roughness,
-        seed: element.seed,
-      });
-    }
-
-    context.globalAlpha = element.opacity / 100;
-    rc.draw(element.shape as Drawable);
-    context.globalAlpha = 1;
-  } else if (element.type === "diamond") {
-    if (!element.shape) {
-      const [
-        topX,
-        topY,
-        rightX,
-        rightY,
-        bottomX,
-        bottomY,
-        leftX,
-        leftY,
-      ] = getDiamondPoints(element);
-      element.shape = generator.polygon(
-        [
-          [topX, topY],
-          [rightX, rightY],
-          [bottomX, bottomY],
-          [leftX, leftY],
-        ],
-        {
+  if (!element.shape) {
+    switch (element.type) {
+      case "rectangle":
+        if (!element.shape) {
```

Comment:
The `if (!element.shape)` check is already done on line 18

Reply:
You're correct. Side effect of copy-pasting :)